### ISSUE TITLE
Add fix_string() to the API

### ISSUE THREAD

**Issue by seanthegeek** [jackdewinter/pymarkdown#1159]
I saw that the CLI has a fix option, but that is not currently exposed in the API.

I'm creating a Python project that uses `html2text` to convert some HTML content to Markdown. The results are pretty messy. The `markdownlint` extension for Visual Studio Code can clean up a lot of linting errors automatically. I'd like to do the same thing before writing the Markdown content to a file to save time and effort.

**Comment by jackdewinter** [comment]
This sounds like a great idea, and we can make sure to work on it.  Can you give me some bad examples of "pretty messy"?

**Comment by seanthegeek** [comment]
Here are two examples of html2text WordPress blog post conversions that sech have over 70 rule violations in the Visual Studio Code markdownlint extension. The editor makes it easy to clean everything up in a few clicks, but if you have a bunch of files, that gets tedious. 

I've observed that the following rules are the ones commonly violated:

- [MD102 - Multiple consecutive blank lines](https://github.com/DavidAnson/markdownlint/blob/v0.34.0/doc/md012.md)
- [MD007 - Unordered list indentation](https://github.com/DavidAnson/markdownlint/blob/v0.34.0/doc/md007.md)
  - There's probably a similar issue with ordered lists, but I didn't see it because I don't use them often :)
- [MD022 - Headings should be surrounded by blank lines](https://github.com/DavidAnson/markdownlint/blob/v0.34.0/doc/md027.md)
- [MD027 - Multiple spaces after blockquote symbol](https://github.com/DavidAnson/markdownlint/blob/v0.34.0/doc/md027.md)
  - This showed up when having lists inside of a blockquote
- [MD026 - Trailing punctuation in heading](https://github.com/DavidAnson/markdownlint/blob/v0.34.0/doc/md026.md)
  - This happened due to a line break in the middle of a long heading (see #424)
- [MD009 - Trailing spaces](https://github.com/DavidAnson/markdownlint/blob/v0.34.0/doc/md009.md)

For what it's worth, I've opened [an issue](https://github.com/Alir3z4/html2text/issues/426) with the html2text project to try and address it there too.

**Comment by seanthegeek** [comment]
This is my project, by the way

https://github.com/seanthegeek/wp2jekyll

**Comment by jackdewinter** [comment]
Just committed changes for you to try out.  Do you need an actual PyPy package release to test it?

**Comment by seanthegeek** [comment]
Unfortunately, I didn't keep the original markdown. Sorry.

**Comment by jackdewinter** [comment]
Sorry, perhaps we should have been clearer.  You want to call PyMarkdown from within your Python application, correct?  If so, the changes are in the main branch.  I am assuming that you need an actual release that you can import from PyPi to use it, correct?

**Comment by seanthegeek** [comment]
Yes

**Comment by seanthegeek** [comment]
If I understand the object returned by the function correctly, I use it like this, right?

```python
from pymarkdown.api import PyMarkdownApi

pymd = PyMarkdownApi()

fix_result = pymd.fix_string(content_markdown)
if fix_result.was_fixed:
    content_markdown = fix_result.fixed_file
````

**Comment by jackdewinter** [comment]
Just released version 0.9.24 which has updated API docs as well as this API.  Please let us know if you have any issues with it.

**Comment by seanthegeek** [comment]
It is working as expected and is now included in wp2jekyll. Thanks!
